scripts/mdbook-forc-documenter/src/plugins.rs

```diff
@@ -28,10 +28,38 @@ pub fn plugin_commands() -> Vec<String> {
             let name = path.file_name().unwrap().to_str().unwrap();
 
             if name.starts_with("forc-") {
-                let plugin = name.split_once('-').unwrap().1;
-                plugins.push(plugin.to_string());
+                // Check for parent plugin (like `forc-client`)
+                let child_plugins = collect_child_plugins(&path.join("Cargo.toml")).unwrap();
+                if child_plugins.is_empty() {
+                    let plugin = name.split_once('-').unwrap().1;
+                    plugins.push(plugin.to_string());
+                } else {
+                    for child_plugin in child_plugins {
+                        let plugin = child_plugin.split_once('-').unwrap().1;
+                        plugins.push(plugin.to_string());
+                    }
```

Comment:
We've already established that `child_plugins` is empty before this branch. It looks like rather than:
 
```rust
                    for child_plugin in child_plugins {
                        let plugin = child_plugin.split_once('-').unwrap().1;
                        plugins.push(plugin.to_string());
                    }
```
this section should be
```rust
                    let plugin = name.split_once('-').unwrap().1;
                    plugins.push(plugin.to_string());
```

Reply:
I am not sure if I understood this change. If `child plugins` is empty we need to add `name.split_once('-').unwrap().1;` part to the plugins same like we did before this PR. If it is not empty, it means that we have a parent plugin that hosts multiple bins just like `forc-client`. So we need to add all child plugins to the list of plugins. I added the check for child plugins so that we can document `forc run` and `forc deploy`. Maybe @bingcicle might have some input here 👀

Reply:
Semantically, I see a plugin as one installable component of the toolchain, so `forc-client` would be the only plugin here instead, with `run` and `deploy` as the available commands to the plugin - it just so happened that for the plugins so far, there was only 1 command available to them. I think I would document them as such since it might potentially be confusing if we documented them as separate plugins, but installed together in `forc-client`.

Perhaps documenting both `forc deploy` and `forc run` under `forc client` would make it clearer that `forc-client` is the installable component here:

```md
...
- 11.4. Plugins
  - 11.4.1 forc client
    - 11.4.1.1 forc deploy
    - 11.4.1.2 forc run
  - 11.4.2 forc explore
  - 11.4.3 forc fmt
  - 11.4.4 forc lsp
```

You might have to update the logic within the forc-documenter script to support sub-chapters though, since it currently naively assumes the commands to run are all unnested.

Reply:
I made the required changes for supporting sub-chapters under plugins to have the documentation structure like the following.

```md
...
- 11.4. Plugins
  - 11.4.1 forc client
    - 11.4.1.1 forc deploy
    - 11.4.1.2 forc run
  - 11.4.2 forc explore
  - 11.4.3 forc fmt
  - 11.4.4 forc lsp
```
First of all, I want to discuss a little bit about the status of `forc-client`. `forc-client` is a plugin that does not produce any executable (unlike `forc-fmt` or any other `forc-*`) once installed. It packs 2 distinct bins, through `[[bins]]` table of its `Cargo.toml`. So we cannot collect any documentation from that, as the document collection phase is basically trying to do `forc-client --help`. So I think we should have the concept of `Virtual Plugins` in the script. If we do not have this kind of separate logic, adding `forc-client` to list of plugins will always result in warning stating that `forc client` has been removed from the toolchain, because there is no `forc-client`. And not adding it to the list of plugins, but having it in the `SUMMARY.md` will also fail. I strongly agree that the structural completeness of the Sway Book requires `forc client` as a chapter under `Plugins` which holds `forc deploy` and `forc run`. So I implemented the logic for supporting these kind of sub-chapters, currently, it supports only a single level of depth, meaning that we cannot have sub-sub-sub chapters 😄. I think if we want to support an arbitrary amount of nesting, we should introduce a graph-like structure to hold this information and introduce a larger refactor to the documenter. Depending on the input from @mitchmindtree and @bingcicle, I can open a separate issue for that as with the current state of the documenter we can create the structure above.

Reply:
Ahh right, makes sense. I think it'd be fair to handle it in a separate issue!

Reply:
> I think if we want to support an arbitrary amount of nesting, we should introduce a graph-like structure to hold this information and introduce a larger refactor to the documenter.

I don't think this should be necessary, I'm happy for us to draw the line at one level of nesting. I think all we're after is a way for a single plugin crate to support more than one command.

@kayagokalp I wonder if we could use something like a `PluginCommand` enum here to make it a little clearer on what's going on here for future maintainers? E.g.

```rust
enum PluginCommand {
    /// The plugin has a single command. Its name matches the plugin.
    Single,
    /// The plugin installs a group of commands with the given names.
    Group(Vec<String>),
}

/// The list of plugins alongside their associated commands.
pub fn plugin_commands() -> BTreeSet<String, PluginCommand> { ... }
```

Reply:
I ended up using a `BTreeMap<String, PluginCommand>` as the return type of the `plugin_commands()` but other than that I created the `PluginCommand` enum and with it I think it is a lot more clearer now, thanks for the review 🙌.
 
Also returning a  `BTreeMap<String, PluginCommand>`, reduced the complexity of the doc addition phase from O(n^2) to O(n). I don't think this is significant as number of plugins is very limited but I enjoyed the fact that it became more readable and more time-efficient at the same time 😄